src/main/java/com/algosenpai/app/UserStats.java

```diff
@@ -0,0 +1,175 @@
+package com.algosenpai.app;
+
+import javafx.util.Pair;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ * Handles temporary storage of user stats while the program is running.
+ * The relationship between the various members is:
+ *
+ * <p>
+ * chapterData: An ArrayList that holds the data for all chapters. Chapters are addressed by index, but can also be
+ * searched by name.
+ * characterImagePath: a string containing the path of the image that is used for the user's character in game.
+ * The two data above are stored permanently to a text file, and retrieved from the text file when the program starts.
+ *</p>
+ *
+ * <p>
+ * currentChapter: A single ChapterStat object that stores the chapter data of the chapter that is currently running.
+ * This is reset for every run. The contents of currentChapter can be transferred to chapterData by calling
+ * saveCurrentChapterToChapterData()
+ * </p>
+ */
+public class UserStats {
+
+    //username of the player
+    private String username;
+    // Array of chapter stats
+    private ArrayList<ChapterStat> chapterData;
+
+    // Stats for the current chapter
+    private ChapterStat currentChapter;
```

Comment:
If the variable names and the purpose of the variable are both trivial, no need to put comments, as they are counter-productive to the reader.